processing/src/main/java/io/druid/segment/data/GenericIndexed.java

```diff
@@ -407,35 +216,56 @@ public T get(int index)
   @Override
   public int indexOf(T value)
   {
-    return bufferIndexed.indexOf(value);
+    return indexOf(this, value);
+  }
+
+  private int indexOf(Indexed<T> indexed, T value)
```

Comment:
Oh, I see, I hadn't realized the change you made to `get()`.  You are changing it so that we are implementing our own polymorphism instead of giving the runtime the ability to optimize it.

Can you do a quick benchmark to make sure that this doesn't adversely affect `get()` performance compared to the old method?  In the large column PR, we explicitly didn't want to touch the abstraction because we didn't want to affect whatever performance was already there.  Given that it seems the bulk of this change is introducing application-enforced polymorphic dispatch to the old code path as well, it would be good to make sure that it's at least a wash in terms of performance (very possible/likely due to branch prediction).